src/pages/POS/POS.vue

```diff
@@ -971,6 +997,122 @@ export default defineComponent({
         });
       }
     },
+    async handleBatchSelected(batchName: string) {
+      if (!this.pendingBatchItem) {
+        return;
+      }
+
+      const { item, quantity } = this.pendingBatchItem;
+      this.pendingBatchItem = null;
+
+      try {
+        const isInventoryItem = await this.fyo.getValue(
+          ModelNameEnum.Item,
+          item.name,
+          'trackItem'
+        );
+        if (!this.batchAddedItems.includes(item.name)) {
+          this.batchAddedItems.push(item.name);
+        }
+
+        let availableQty = 0;
+        if (isInventoryItem) {
+          availableQty =
+            (await fyo.db.getStockQuantity(
+              item.name,
+              undefined,
+              undefined,
+              undefined,
+              batchName
+            )) ?? 0;
+
+          const itemIndex = this.items.findIndex((i) => i.name === item.name);
+          if (itemIndex !== -1) {
+            this.items[itemIndex].availableQty = availableQty ?? 0;
+          }
+        }
+
+        const existingItems =
+          this.sinvDoc.items?.filter(
+            (invoiceItem) =>
+              invoiceItem.item === item.name &&
+              invoiceItem.batch === batchName &&
+              !invoiceItem.isFreeItem
+          ) ?? [];
+
+        const completeItem = (await this.fyo.doc.getDoc(
+          ModelNameEnum.Item,
+          item.name
+        )) as Item;
+
+        await validateQty(
+          this.sinvDoc as SalesInvoice,
+          completeItem,
+          existingItems as InvoiceItem[]
+        );
```

Comment:
```suggestion
        await validateQty(
          this.sinvDoc as SalesInvoice,
          itemDoc,
          existingItems as InvoiceItem[]
        );
```

<br/>

itemDoc is already available - use it here